scripts/Diffraction/single_crystal/sxd.py

```diff
@@ -0,0 +1,320 @@
+            foc_spec = banks_foc.getSpectrum(igrp)
+            ispecs = ws.getIndicesFromDetectorIDs(list(foc_spec.getDetectorIDs()))
+            for ispec in ispecs:
+                # assume lambda dep. efficiency differs only by const scale factor between detectors on same bank
+                # also assume that bg counts are subject to same efficiency
+                scale = ws.readY(int(ispec)).sum() / ybank_sum
+                ws.setY(int(ispec), scale * ybank)
+                ws.setE(int(ispec), scale * ebank)
+        mantid.DeleteWorkspace("grouped")
+
+    def process_vanadium(self):
+        # load empty and vanadium
+        empty_ws = self.load_run(self.empty_runno)
+        self.van_ws = self.load_run(self.van_runno)
+        # create grouping file per bank
+        self.grp_ws, _, self.ngrp = mantid.CreateGroupingWorkspace(
+            InputWorkspace=self.van_ws, GroupDetectorsBy="bank", OutputWorkspace="bank_groups", EnableLogging=False
+        )
+        self._minus_workspaces(self.van_ws, empty_ws)
+        self._replace_spectra_with_focussed_bank(self.van_ws)
+        # correct vanadium for absorption
+        mantid.SetSample(
+            self.van_ws,
+            Geometry={"Shape": "CSG", "Value": self.sphere_shape},
+            Material={"ChemicalFormula": "V0.95-Nb0.05", "SampleNumberDensity": 0.0722},
+            EnableLogging=False,
+        )
+        mantid.ConvertUnits(InputWorkspace=self.van_ws, OutputWorkspace=self.van_ws, Target="Wavelength", EnableLogging=False)
+        vanadium_transmission = mantid.SphericalAbsorption(
+            InputWorkspace=self.van_ws, OutputWorkspace="vanadium_transmission", EnableLogging=False
+        )
+        self._divide_workspaces(self.van_ws, vanadium_transmission)
+        mantid.DeleteWorkspace(vanadium_transmission, EnableLogging=False)
+        mantid.ConvertUnits(InputWorkspace=self.van_ws, OutputWorkspace=self.van_ws, Target="TOF", EnableLogging=False)
+
+    @staticmethod
+    def flip_bank_1(ws):
+        wsname = SXD.retrieve(ws).name()
+        mantid.MoveInstrumentComponent(
+            Workspace=wsname, ComponentName="bank1", RelativePosition=False, EnableLogging=False
+        )  # move to origin
+        mantid.RotateInstrumentComponent(Workspace=wsname, ComponentName="bank1", Y=1, Angle=37.5, EnableLogging=False)
+        mantid.RotateInstrumentComponent(Workspace=wsname, ComponentName="bank1", X=1, Y=1, Angle=180, EnableLogging=False)
+        mantid.RotateInstrumentComponent(Workspace=wsname, ComponentName="bank1", X=1, Angle=-37.5, EnableLogging=False)
+        mantid.MoveInstrumentComponent(
+            Workspace=wsname, ComponentName="bank1", X=0.136971, Z=-0.178505, RelativePosition=False, EnableLogging=False
+        )
+
+    @staticmethod
+    def calibrate_sxd_panels(ws, peaks, save_dir, tol=0.15, maxShiftInMeters=0.025):
+        """
+        Calibrate SXD panels and apply the calibration to th
```

Comment:
Given this is a fairly chunky function it's worth filling out the description here